src/librustc/mir/visit.rs

```diff
@@ -786,3 +804,58 @@ pub enum LvalueContext<'tcx> {
     StorageLive,
     StorageDead,
 }
+
+impl<'tcx> LvalueContext<'tcx> {
+    pub fn is_storage_marker(&self) -> bool {
+        match *self {
+            LvalueContext::StorageLive | LvalueContext::StorageDead => true,
+            _ => false,
+        }
+    }
+
+    pub fn is_storage_live_marker(&self) -> bool {
+        match *self {
+            LvalueContext::StorageLive => true,
+            _ => false,
+        }
+    }
+
+    pub fn is_storage_dead_marker(&self) -> bool {
+        match *self {
+            LvalueContext::StorageDead => true,
+            _ => false,
+        }
+    }
+
+    /// Returns true if this lvalue context represents a mutation (def).
+    pub fn is_def(&self) -> bool {
```

Comment:
See e.g. http://www.cs.cmu.edu/afs/cs/academic/class/15745-s05/www/lectures/lect2.pdf — "def" seems to be the right term. @nikomatsakis, want to weigh in?